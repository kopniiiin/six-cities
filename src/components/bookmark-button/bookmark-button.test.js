import React from "react";
import {shallow} from "enzyme";

import BookmarkButton, {testProps} from "./bookmark-button.jsx";

describe(`snapshot test: BookmarkButton component`, () => {
  it(`should render correctly`, () => expect(
      shallow(<BookmarkButton {...testProps}/>)
  ).toMatchSnapshot());

  it(`should render active`, () => expect(
      shallow(<BookmarkButton {...Object.assign({}, testProps, {isActive: true})}/>)
  ).toMatchSnapshot());

  it(`should render big`, () => expect(
      shallow(<BookmarkButton {...Object.assign({}, testProps, {isBig: true})}/>)
  ).toMatchSnapshot());
});
