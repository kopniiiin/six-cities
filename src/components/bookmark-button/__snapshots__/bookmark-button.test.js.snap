// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: BookmarkButton component should render active 1`] = `
<button
  className="block__bookmark-button block__bookmark-button--active button"
  type="button"
>
  <svg
    className="block__bookmark-icon"
    height={19}
    width={18}
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    In bookmarks
  </span>
</button>
`;

exports[`snapshot test: BookmarkButton component should render big 1`] = `
<button
  className="block__bookmark-button  button"
  type="button"
>
  <svg
    className="block__bookmark-icon"
    height={33}
    width={31}
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To bookmarks
  </span>
</button>
`;

exports[`snapshot test: BookmarkButton component should render correctly 1`] = `
<button
  className="block__bookmark-button  button"
  type="button"
>
  <svg
    className="block__bookmark-icon"
    height={19}
    width={18}
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To bookmarks
  </span>
</button>
`;
